### ISSUE TITLE
Forkserver sometimes seems to crash in qemu mode on aarch64 (maybe others)?

### ISSUE THREAD

**Issue by vanhauser-thc** [AFLplusplus/AFLplusplus#1550]
@WorksButNotTested bugfix [fix qemu_mode speed bug](https://github.com/AFLplusplus/AFLplusplus/commit/5e04c9693b19038871f3aa93544cac187785bc1a) introduced an issue in the forkserver it seems. It does not happen on all targets, I accidently encountered it when testing some other issues.

```
aarch64-linux-gnu-gcc -static -o test-instr test-instr.c
AFL_DEBUG=1 AFL_PATH=`pwd` ./afl-fuzz -Q -i in -o out ./test-instr
[...]
[*] Target map size: 65536
[D] DEBUG: calibration stage 1/7
Neither one or zero? How quaint!
[D] DEBUG: calibration stage 2/7
qemu: uncaught target signal 11 (Segmentation fault) - core dumped

[-] Oops, the program crashed with one of the test cases provided. There are
    several possible explanations:
[...]
```

@WorksButNotTested can you please have a look?

because it could be an issue with page layout, addresses, whatever, I have attached the compiled binary.
[test-instr.zip](https://github.com/AFLplusplus/AFLplusplus/files/9747549/test-instr.zip)

**Comment by WorksButNotTested** [comment]
As this issue only occurs when running in the fuzzer, and only after the flush has been completed (rather than during it). I removed qemuafl's signal handlers using:
`signal(SIGSEGV, SIG_DFL)`
and managed to generate this core dump ([crash.zip](https://github.com/AFLplusplus/AFLplusplus/files/9762216/crash.zip)). I have included the binary too. However, the crash appears to be from executing code outside of any of the loaded modules:
`info proc mappings` and the stack similarly seems full of these out-of-bounds addresses.

So my best guess would be that it has crashed horribly when trying to execute some TCG generated code!

@domenukk do you have any insights on where to go from here?

**Comment by WorksButNotTested** [comment]
Ok. So I've tried a different approach here...
https://github.com/AFLplusplus/qemuafl/pull/40

I flush the translation cache from the host context (when translating the entry point) rather than from the guest context when executing it. This seems to avoid the flush being deferred as well as the gnarly translation cache corruption issues we were seeing.

It seems to fix both the aarch64 crash above and still deal with the original problem I was trying to fix as demonstrated by my t3 test at the bottom of here...
https://github.com/AFLplusplus/AFLplusplus/issues/1540

Lastly, it doesn't seem to re-introduce the 10x performance regressions from 4.02c we saw with 4.03c.

Let's hope this one is better!?!
